kolibri/core/logger/csv_export.py

```diff
@@ -103,7 +102,35 @@ def cache_content_title(obj):
     )
 )
 
-map_object = partial(output_mapper, labels=labels, output_mappings=mappings)
+
+def get_max_ancestor_depth():
+    max_depth = 0
+    for node in ContentNode.objects.filter(
+        content_id__in=ContentSummaryLog.objects.values_list("content_id", flat=True)
+    ):
+        max_depth = max(max_depth, len(node.ancestors))
+    return max_depth
+
+
+def add_ancestors_info(row, ancestors, max_depth):
+    row.update(
+        {
+            f"Topic level {level + 1}": ancestors[level]["title"]
+            if level < len(ancestors)
+            else ""
+            for level in range(max_depth)
+        }
+    )
+
+
+def map_object(item):
+    mapped_item = output_mapper(item, labels=labels, output_mappings=mappings)
+    node = ContentNode.objects.filter(content_id=item["content_id"]).first()
```

Comment:
We could halve the number of queries here by getting and caching the ancestor information when we get the title too.

So perhaps we have a general helper function `cache_content_data` that is used by a renamed `get_content_title` and a new `get_content_ancestors` - each one calls the `cache_content_data` function that queries for the content node and stores both the title and the ancestors in the cache and returns them.

We could also use fill this cache in get_max_ancestor_depth calculation, so that it is already populated, as we've had to iterate over every content node already to calculate this.